angr/sim_type.py

```diff
@@ -1079,19 +1097,43 @@ def parse_type(defn, preprocess=True):
     if pycparser is None:
         raise ImportError("Please install pycparser in order to parse C definitions")
 
-    defn = 'typedef ' + defn.strip('; \n\t\r') + ' QQQQ;'
+    defn = re.sub(r"/\*.*?\*/", r"", defn)
 
-    if preprocess:
-        defn = do_preprocess(defn)
+    parser = pycparser.CParser()
+
+    # pycparser hack to parse type expressions
+    errorlog = logging.getLogger('yacc logger')
+    errorlog.setLevel(logging.ERROR)
+    parser.cparser = pycparser.ply.yacc.yacc(module=parser,
+                                             start='parameter_declaration',
+                                             debug=False,
+                                             optimize=False,
+                                             errorlog=errorlog)
 
-    node = pycparser.c_parser.CParser().parse(make_preamble()[0] + defn)
-    if not isinstance(node, pycparser.c_ast.FileAST) or \
-            not isinstance(node.ext[-1], pycparser.c_ast.Typedef):
+    node = parser.parse(text=defn, scope_stack=_make_scope())
+    if not isinstance(node, pycparser.c_ast.Typename):
         raise ValueError("Something went horribly wrong using pycparser")
 
-    decl = node.ext[-1].type
+    decl = node.type
     return _decl_to_type(decl)
 
+
+def _accepts_scope_stack():
+    """
+    pycparser hack to include scope_stack as parameter in CParser parse method
+    """
+    def parse(self, text, scope_stack=[dict()], filename='', debuglevel=0):
```

Comment:
The values that are created when you write expressions as default arguments are evaluated at declaration time, not call time, so this mutable value will be stored by reference as the default argument for all calls, so future calls will receive a mutated default value. The correct way to do things like this is `scope_stack=None`, and then `self._scope_stack = [dict()] if scope_stack is None else scope_stack`.

Reply:
I have corrected this. Thanks for the information!